### DISCUSSION TITLE
Feature request: "clickable" prop for QIcon

### DISCUSSION THREAD

**Discussion by iliubinskii** [quasarframework/quasar#12236] [Ideas / Proposals] [answered]
Clickable QIcon can be used in QInput append and prepend.
It can share hover effect with QInput's "clearable" icon.

**Comment by metalsadman** [comment, marked as answer]
just give it `@click` event and maybe add a class `cursor-pointer`/(a custom class of your own) to indicate that it's clickable when hovered.

**Comment by iliubinskii** [comment]
Thx for response.

I am already using @click and cursor-pointer.

However, I have difficulty with making hover effect exactly the same as in QInput's "clearable" icon.

Consilder this sample:
https://codepen.io/ilyalyu/pen/oNojjKP?editors=101

I know how to set QIcon's color and how to add hover effect.
I can also take screenshot or use F12 to find out main color and hover color of "clearable" icon.
In my sample I applied grey-8 color to one icon which is close to "clearable" icon's main color.

But imagine that after that I apply "dark" attribute (second input in my sample).
In, this case my grey-8 icon falls out of design and I have to adapt it's colors again.

Now imagine that I edit quasar color variables:
https://quasar.dev/style/sass-scss-variables
Will I have to adapt my manual color and hover effect again?

It would be much more consistent to have "clickable" prop that would warant uniform behavior of my QIcon and QInput's "clearable" icon in all possible circumstances.
